discounts/rust/discounts/default/src/run.rs

```diff
@@ -0,0 +1,421 @@
+            selection_strategy: DeliveryDiscountSelectionStrategy::ALL,
+            candidates,
+        })],
+    })
+}
+
+#[shopify_function_target(
+    target = "cart_fetch",
+    query_path = "src/fetch.graphql",
+    schema_path = "schema.graphql"
+)]
+fn cart_fetch(input: CartFetchResponseData) -> Result<FunctionCartFetchResult> {
+    let body = json!({
+        "buyerIdentity": {
+            "email": input.cart.buyer_identity.as_ref().unwrap().email
+        }
+    });
+
+    Ok(FunctionCartFetchResult {
+        request: Some(CartHttpRequest {
+            body: Some(body.to_string()),
+            headers: vec![],
+            json_body: Some(body.clone()),
+            method: CartHttpRequestMethod::POST,
+            policy: CartHttpRequestPolicy {
+                read_timeout_ms: 2000,
+            },
+            url: "https://example.com".to_string(),
+        }),
+    })
+}
+
+#[shopify_function_target(
+    target = "delivery_fetch",
+    query_path = "src/fetch.graphql",
+    schema_path = "schema.graphql"
+)]
+fn delivery_fetch(input: DeliveryFetchResponseData) -> Result<FunctionDeliveryFetchResult> {
+    let body = json!({
+        "buyerIdentity": {
+            "email": input.cart.buyer_identity.as_ref().unwrap().email
+        }
+    });
+
+    Ok(FunctionDeliveryFetchResult {
+        request: Some(DeliveryHttpRequest {
+            body: Some(body.to_string()),
+            headers: vec![],
+            json_body: Some(body.clone()),
+            method: DeliveryHttpRequestMethod::POST,
+            policy: DeliveryHttpRequestPolicy {
+                read_timeout_ms: 2000,
+            },
+            url: "https://example.com".to_string(),
+        }),
+    })
+}
+
+fn create_order_discount(metafield: &Metafield) -> CartOperation {
+    CartOperation::AddOrderDiscounts(OrderDiscounts {
+        selection_strategy: OrderDiscountSelectionStrategy::FIRST,
+        candidates: vec![OrderDiscountCandidate {
+            targets: vec![OrderDiscountCandidateTarget::OrderSubtotal(
+                OrderSubtotalTarget {
+                    excluded_variant_ids: vec![],
+                },
+            )],
+            associated_discount_code: None,
+            message: None,
+            value: OrderDiscountCandidateValue::Percentage(CartPercentage {
+                value: metafield.cart_percent.unwrap(),
+            }),
+            conditions: None,
+        }],
+    })
+}
+
+fn create_product_discount(cart_line_id: &str, metafield: &Metafield) -> CartOperation {
+    CartOperation::AddProductDiscounts(ProductDiscounts {
+        selection_strategy: ProductDiscountSelectionStrategy::FIRST,
+        candidates: vec![ProductDiscountCandidate {
+            targets: vec![ProductDiscountCandidateTarget::CartLine(CartLineTarget {
+                id: cart_line_id.to_string(),
+                quantity: Some(1),
+            })],
+            associated_discount_code: None,
```

Comment:
If we add enteredDiscountCode validation in the Fetch function, then we could include an associated_discount_code from the valid discount codes returned by the Fetch.